Developer Settings appear when API is down Stable/Beta

This should be avoided

![screenshot 2017-04-12 20 23 39](https://cloud.githubusercontent.com/assets/4798491/24987569/ee2c4b56-1fbd-11e7-9cef-51dcac4bb8a7.png)

```
DIM v3.18.0.851 (beta) - Please report any errors to https://www.reddit.com/r/destinyitemmanager
angular.js:14525 Error: No Destiny account was found on PlayStation. Do you have the right platform selected?
    at l (dimBungieService.factory.js:101)
    at u (angular.js:16832)
    at angular.js:16876
    at p.$digest (angular.js:17971)
    at p.$apply (angular.js:18269)
    at u (angular.js:12387)
    at g (angular.js:12613)
    at XMLHttpRequest.x.onload (angular.js:12541) "Possibly unhandled rejection: {}"
(anonymous) @ angular.js:14525
dimBungieService.factory.js:371 Failed to load character progression Error: Time out during communication with the game servers.  Please try again later.
    at l (dimBungieService.factory.js:113)
    at u (angular.js:16832)
    at angular.js:16876
    at p.$digest (angular.js:17971)
    at p.$apply (angular.js:18269)
    at u (angular.js:12387)
    at g (angular.js:12613)
    at XMLHttpRequest.x.onload (angular.js:12541)
(anonymous) @ dimBungieService.factory.js:371
www.bungie.net/Platform/Destiny/2/Account/4611686018461196906/Character/2305843009399001197/Inventory/?definitions=false Failed to load resource: net::ERR_CONNECTION_RESET
angular.js:14525 Error: You may not be connected to the internet.
    at l (dimBungieService.factory.js:59)
    at u (angular.js:16832)
    at angular.js:16876
    at p.$digest (angular.js:17971)
    at p.$apply (angular.js:18269)
    at u (angular.js:12387)
    at g (angular.js:12613)
    at XMLHttpRequest.$ (angular.js:12551) "Possibly unhandled rejection: {}"
(anonymous) @ angular.js:14525
www.bungie.net/Platform/Destiny/2/Account/4611686018461196906/Character/2305843009399001212/Inventory/?definitions=false Failed to load resource: net::ERR_CONNECTION_RESET
www.bungie.net/Platform/Destiny/2/Account/4611686018461196906/Character/2305843009399027747/Advisors/V2/?definitions=false Failed to load resource: net::ERR_CONNECTION_RESET
main-2f767c.js:1 Failed to load advisors Error: You may not be connected to the internet.
    at l (main-2f767c.js:1)
    at u (main-2f767c.js:1)
    at main-2f767c.js:1
    at p.$digest (main-2f767c.js:1)
    at p.$apply (main-2f767c.js:1)
    at u (main-2f767c.js:1)
    at g (main-2f767c.js:1)
    at XMLHttpRequest.$ (main-2f767c.js:1)
```

--- a/src/index.js
+++ b/src/index.js
@@ -78,7 +78,9 @@ require('./scripts/minmax/dimMinMaxItem.directive');
 require('./scripts/minmax/dimMinMaxLocks.directive');
 require('./scripts/minmax/dimMinMaxCharSelect.directive');
 require('./scripts/debug/dimDebugItem.controller');
-require('./scripts/developer/dimDeveloper.controller');
+if ($DIM_FLAVOR === 'dev') {
+  require('./scripts/developer/dimDeveloper.controller');
+}
 require('./scripts/materials-exchange/dimCollapsible.directive');
 require('./scripts/login/dimLogin.controller');
 

--- a/src/scripts/dimApp.routes.js
+++ b/src/scripts/dimApp.routes.js
@@ -1,7 +1,6 @@
 import best from 'app/views/best.template.html';
 import materialExchange from 'app/views/mats-exchange.template.html';
 import debugItem from 'app/views/debugItem.template.html';
-import developer from 'app/scripts/developer/developer.template.html';
 import login from 'app/scripts/login/login.template.html';
 
 function routes($stateProvider, $urlRouterProvider) {
@@ -26,16 +25,20 @@ function routes($stateProvider, $urlRouterProvider) {
     url: '/debugItem/:itemId',
     templateUrl: debugItem
   }, {
-    name: 'developer',
-    parent: 'content',
-    url: '/developer',
-    templateUrl: developer
-  }, {
     name: 'login',
     parent: 'shell',
     url: '/login',
     templateUrl: login
   }];
+
+  if ($DIM_FLAVOR === 'dev') {
+    states.push({
+      name: 'developer',
+      parent: 'content',
+      url: '/developer',
+      templateUrl: require('app/scripts/developer/developer.template.html')
+    });
+  }
 
   states.forEach((state) => {
     $stateProvider.state(state);

--- a/src/scripts/oauth/oauth.module.js
+++ b/src/scripts/oauth/oauth.module.js
@@ -5,11 +5,9 @@ require('angular-local-storage');
 angular.module('dim-oauth', ['LocalStorageModule'])
   .run(function($rootScope, $state) {
     $rootScope.$on('dim-no-token-found', function() {
-      if ($DIM_FLAVOR === 'release' || $DIM_FLAVOR === 'beta') {
+      if ($DIM_FLAVOR !== 'dev') {
         $state.go('login');
-        return;
-      }
-      if (!localStorage.apiKey || !localStorage.authorizationURL) {
+      } else if (!localStorage.apiKey || !localStorage.authorizationURL) {
         $state.go('developer');
       } else {
         $state.go('login');

--- a/src/scripts/oauth/oauth.service.js
+++ b/src/scripts/oauth/oauth.service.js
@@ -15,11 +15,22 @@ function OAuthService($q, $injector, localStorageService, OAuthTokenService) {
   function refreshToken() {
     const $http = $injector.get('$http');
 
+    var apiKey;
+    if ($DIM_FLAVOR === 'release' || $DIM_FLAVOR === 'beta') {
+      if (window.chrome && window.chrome.extension) {
+        apiKey = $DIM_API_KEY;
+      } else {
+        apiKey = $DIM_WEB_API_KEY;
+      }
+    } else {
+      apiKey = localStorageService.get('apiKey');
+    }
+
     return $http({
       method: 'POST',
       url: 'https://www.bungie.net/Platform/App/GetAccessTokensFromRefreshToken/',
       headers: {
-        'X-API-Key': localStorageService.get('apiKey')
+        'X-API-Key': apiKey
       },
       data: {
         refreshToken: OAuthTokenService.getRefreshToken().value

--- a/src/scripts/services/dimBungieService.factory.js
+++ b/src/scripts/services/dimBungieService.factory.js
@@ -103,8 +103,12 @@ function BungieService($rootScope, $q, $timeout, $http, $state, dimState, toaste
     case 2101: // ApiInvalidOrExpiredKey
     case 2102: // ApiKeyMissingFromRequest
     case 2107: // OriginHeaderDoesNotMatchKey
-      $state.go('developer');
-      return $q.reject(new Error($translate.instant('BungieService.DevVersion')));
+      if ($DIM_FLAVOR === 'dev') {
+        $state.go('developer');
+        return $q.reject(new Error($translate.instant('BungieService.DevVersion')));
+      } else {
+        return $q.reject(new Error($translate.instant('BungieService.Difficulties')));
+      }
     }
 
     // Any other error

